metaflow/plugins/conda_escape/__init__.py

```diff
@@ -0,0 +1,162 @@
+# Portions of this plugin are inspired by RPyC
+# https://rpyc.readthedocs.io/
+#
+# The license for that software is reproduced below
+#
+# Copyright (c) 2005-2013
+#  Tomer Filiba (tomerfiliba@gmail.com)
+#  Copyrights of patches are held by their respective submitters
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import os
+import sys
+from subprocess import Popen, PIPE
+
+from itertools import chain
+
+from .exception_transferer import RemoteInterpreterException
+from .client_modules import create_modules
+
+# Determine what is the python executable to use for the escape hatch. To do this,
+# we look for ESCAPE_HATCH_PY in the environment AND STORE IT. When metaflow
+# is first launched by the user, it is in the outside non-metaflow-created-conda
```

Comment:
I am not sure I follow here - what does `it` refer to - the flow's interpreter? And what does `non-metaflow-created-conda` environment refer to?